src/Pangraph/Internal/HexmlExtra.hs

```diff
@@ -5,24 +5,51 @@ import Data.ByteString (ByteString)
 import Text.XML.Hexml
 import Pangraph
 
--- * A module containing some exclusivly Hexml helper functions and some which have common interfacing functions.
 
--- | Find the ['Node'] with the final in the ['ByteString'] after following the
--- 'Node' names recursively.
+-- * A module containing containing extra Hexml functions as well as functions to ease conversions between the libraries.
+
+type HexmlAttribute = Text.XML.Hexml.Attribute
+type PangraphAttribute = Pangraph.Attribute
+
+-- | Find the ['Node'] with the final name in the ['ByteString'] after following the 'Node' names recursively, including the root node's name.
+-- Given `[]` returns '[]'.
 followChildren :: Node -> [ByteString] -> [Node]
-followChildren h [] = [h]
-followChildren h bs = (concatMap recurse . childrenBy h) (head bs)
+followChildren _ [] = []
+followChildren h bs = if head bs == name h
```

Comment:
It may be easier to read this if you use guard expressions, e.g. `| head bs == ...`. Even better, avoid using partial functions like `head` by pattern-matching (otherwise a future refactoring could introduce a bug).